docs/conf.py

```diff
@@ -21,6 +21,39 @@
 
 import more_itertools
 
+# -- Preprocess README.rst -----------------------------------------------------
+
+import re
+
+with open('../README.rst', 'r+') as source:
+    readme_file = source.readlines()
+
+# Make sure build directory exists. This is required for github workflow.
+build_dir = '_build'
+if not os.path.exists(build_dir):
+    os.makedirs(build_dir)
+
+# Change absolute links in README.rst to relative ones. This way the online
+# documentation copmiles with relative links, while README.rst on GitHub can
+# have absolute links. See issue #551.
+root_path = 'https://more-itertools.readthedocs.io/en/stable/'
+in_table = False
+with open(os.path.join('.', build_dir, 'README.pprst'), 'w+') as target:
```

Comment:
Why is the `w+` needed here?

Reply:
apparently it's not. I've changed it to 'w'